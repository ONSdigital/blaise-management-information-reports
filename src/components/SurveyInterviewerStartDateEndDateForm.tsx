import React, { ReactElement } from "react";
import { StyledForm } from "blaise-design-system-react-components";
import { EndDateField, InterviewerField, StartDateField, SurveyField } from "./FormFields";

interface Props {
    interviewer: string | undefined
    surveyTLA: string | undefined
    startDate: Date
    endDate: Date
    onSubmit: (values: any, setSubmitting: (isSubmitting: boolean) => void) => void;
}

const SurveyInterviewerStartDateEndDateForm = ({
    interviewer,
    surveyTLA,
    startDate,
    endDate,
    onSubmit
}: Props): ReactElement => {

    const fields = [
        SurveyField(surveyTLA),
        InterviewerField(interviewer),
        StartDateField(startDate),
        EndDateField(endDate),
    ];

    return (
        <><StyledForm fields={fields} onSubmitFunction={onSubmit} submitLabel={"Next"}/></>
    );
};

export default SurveyInterviewerStartDateEndDateForm;
