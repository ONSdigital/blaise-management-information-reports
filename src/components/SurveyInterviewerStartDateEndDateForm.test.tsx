/**
 * @jest-environment jsdom
 */

import "@testing-library/jest-dom";
import { createMemoryHistory } from "history";
import { render } from "@testing-library/react";
import { Router } from "react-router";
import { act } from "react-dom/test-utils";
import { screen } from "@testing-library/dom";
import React from "react";
import MockDate from "mockdate";
import SurveyInterviewerStartDateEndDateForm from "./SurveyInterviewerStartDateEndDateForm";
import flushPromises from "../tests/utilities";

const christmasEve97 = "1997-12-24";

describe("form - survey, interviewer, start date, end date", () => {
    beforeEach(() => {
        MockDate.set(new Date(christmasEve97));
    });

    afterEach(() => {
        MockDate.reset();
    });

    it("matches snapshot", async () => {
        const history = createMemoryHistory();
        const wrapper = render(
            <Router history={history}>
                <SurveyInterviewerStartDateEndDateForm
                    interviewer=""
                    surveyTLA=""
                    startDate={new Date()}
                    endDate={new Date()}
                    onSubmit={() => true}
                />
            </Router>,
        );

        await act(async () => {
            await flushPromises();
        });

        expect(await wrapper).toMatchSnapshot();
    });

    it("renders correctly", async () => {
        const history = createMemoryHistory();
        await act(async () => {
            render(
                <Router history={history}>
                    <SurveyInterviewerStartDateEndDateForm
                        interviewer=""
                        surveyTLA=""
                        startDate={new Date()}
                        endDate={new Date()}
                        onSubmit={() => true}
                    />
                </Router>,
            );
        });
        expect(screen.queryByText("Select survey")).toBeVisible();
        expect(screen.queryByText("Show all surveys")).toBeVisible();
        expect(screen.queryByText("LMS")).toBeVisible();
        expect(screen.queryByText("Labour Market Survey")).toBeVisible();
        expect(screen.queryByText("OPN")).toBeVisible();
        expect(screen.queryByText("Opinions and Lifestyle Survey")).toBeVisible();
        expect(screen.queryByText("Interviewer ID")).toBeVisible();
        expect(screen.queryByText("Start date")).toBeVisible();
        expect(screen.queryByText("End date")).toBeVisible();
        expect(screen.queryByText("Next")).toBeVisible();
    });
});
