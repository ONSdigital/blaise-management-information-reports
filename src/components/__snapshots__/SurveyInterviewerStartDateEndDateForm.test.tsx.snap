// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form - survey, interviewer, start date, end date matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        action="#"
      >
        <fieldset
          class="ons-fieldset"
        >
          <legend
            class="ons-fieldset__legend"
          >
            Select survey
          </legend>
          <div
            class="ons-radios__items"
            id="Survey TLA"
          >
            <p
              class="ons-radios__item"
            >
              <span
                class="ons-radio"
              >
                <input
                  checked=""
                  class="ons-radio__input js-radio"
                  id="all"
                  initial_value=""
                  name="Survey TLA"
                  type="radio"
                  value=""
                />
                <label
                  class="ons-radio__label "
                  for="all"
                  id="all-label"
                >
                  Show all surveys
                </label>
              </span>
            </p>
            <br />
            <p
              class="ons-radios__item"
            >
              <span
                class="ons-radio"
              >
                <input
                  class="ons-radio__input js-radio"
                  id="lms"
                  initial_value=""
                  name="Survey TLA"
                  type="radio"
                  value="lms"
                />
                <label
                  class="ons-radio__label ons-label--with-description"
                  for="lms"
                  id="lms-label"
                >
                  LMS
                  <span
                    class="ons-label__description ons-radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Labour Market Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
            <p
              class="ons-radios__item"
            >
              <span
                class="ons-radio"
              >
                <input
                  class="ons-radio__input js-radio"
                  id="opn"
                  initial_value=""
                  name="Survey TLA"
                  type="radio"
                  value="opn"
                />
                <label
                  class="ons-radio__label ons-label--with-description"
                  for="opn"
                  id="opn-label"
                >
                  OPN
                  <span
                    class="ons-label__description ons-radio__label--with-description"
                    id="white-label-description-hint"
                  >
                    Opinions and Lifestyle Survey
                  </span>
                </label>
              </span>
            </p>
            <br />
          </div>
        </fieldset>
        <div
          class="ons-field"
        >
          <label
            class="ons-label "
            for="Interviewer ID"
          >
            Interviewer ID
          </label>
          <input
            id="Interviewer ID"
            initial_value=""
            name="Interviewer ID"
            type="text"
            value=""
          />
        </div>
        <div
          class="ons-field"
        >
          <label
            class="ons-label "
            for="Start date"
          >
            Start date
          </label>
          <input
            id="Start date"
            initial_value="1997-12-24"
            name="Start date"
            type="date"
            value="1997-12-24"
          />
        </div>
        <div
          class="ons-field"
        >
          <label
            class="ons-label "
            for="End date"
          >
            End date
          </label>
          <input
            id="End date"
            initial_value="1997-12-24"
            name="End date"
            type="date"
            value="1997-12-24"
          />
        </div>
        <br />
        <button
          class="ons-btn "
          data-testid="submit-button"
          type="submit"
        >
          <span
            class="ons-btn__inner"
          >
            Next
          </span>
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      action="#"
    >
      <fieldset
        class="ons-fieldset"
      >
        <legend
          class="ons-fieldset__legend"
        >
          Select survey
        </legend>
        <div
          class="ons-radios__items"
          id="Survey TLA"
        >
          <p
            class="ons-radios__item"
          >
            <span
              class="ons-radio"
            >
              <input
                checked=""
                class="ons-radio__input js-radio"
                id="all"
                initial_value=""
                name="Survey TLA"
                type="radio"
                value=""
              />
              <label
                class="ons-radio__label "
                for="all"
                id="all-label"
              >
                Show all surveys
              </label>
            </span>
          </p>
          <br />
          <p
            class="ons-radios__item"
          >
            <span
              class="ons-radio"
            >
              <input
                class="ons-radio__input js-radio"
                id="lms"
                initial_value=""
                name="Survey TLA"
                type="radio"
                value="lms"
              />
              <label
                class="ons-radio__label ons-label--with-description"
                for="lms"
                id="lms-label"
              >
                LMS
                <span
                  class="ons-label__description ons-radio__label--with-description"
                  id="white-label-description-hint"
                >
                  Labour Market Survey
                </span>
              </label>
            </span>
          </p>
          <br />
          <p
            class="ons-radios__item"
          >
            <span
              class="ons-radio"
            >
              <input
                class="ons-radio__input js-radio"
                id="opn"
                initial_value=""
                name="Survey TLA"
                type="radio"
                value="opn"
              />
              <label
                class="ons-radio__label ons-label--with-description"
                for="opn"
                id="opn-label"
              >
                OPN
                <span
                  class="ons-label__description ons-radio__label--with-description"
                  id="white-label-description-hint"
                >
                  Opinions and Lifestyle Survey
                </span>
              </label>
            </span>
          </p>
          <br />
        </div>
      </fieldset>
      <div
        class="ons-field"
      >
        <label
          class="ons-label "
          for="Interviewer ID"
        >
          Interviewer ID
        </label>
        <input
          id="Interviewer ID"
          initial_value=""
          name="Interviewer ID"
          type="text"
          value=""
        />
      </div>
      <div
        class="ons-field"
      >
        <label
          class="ons-label "
          for="Start date"
        >
          Start date
        </label>
        <input
          id="Start date"
          initial_value="1997-12-24"
          name="Start date"
          type="date"
          value="1997-12-24"
        />
      </div>
      <div
        class="ons-field"
      >
        <label
          class="ons-label "
          for="End date"
        >
          End date
        </label>
        <input
          id="End date"
          initial_value="1997-12-24"
          name="End date"
          type="date"
          value="1997-12-24"
        />
      </div>
      <br />
      <button
        class="ons-btn "
        data-testid="submit-button"
        type="submit"
      >
        <span
          class="ons-btn__inner"
        >
          Next
        </span>
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
